docs/DeveloperGuide.md

```diff
@@ -1,25 +1,277 @@
 # Developer Guide
 
-## Design & implementation
+## Design
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### Application Architecture
+![Architecture](images-dg/Architecture.png)
 
+The **Architecture Diagram** given above explains the high-level design of the Flash Card Manager Application.
+
+`Main` is responsible for initializing the other components in the program and linking them up correctly.
+
+* `Ui` Takes in instructions from user and displays the output to the user
+* `Logic` Consists of the `Parsers` and the `Commands`. The `Parser` decipher the user input and executes the specific `Command` that affects the change the user wishes.
+* `Model` Holds the data that is in memory as the program runs. It consists of the 3 components:
+    * `Flash Card` : A single question and answer pair.
+    * `Deck` : A list of `Flash Card`s under a common topic.
+    * `Deck List` : A complete list of all the `Deck`s in memory.
+* `Storage` Reads and writes data from and to a text file.
+
+#### How to **components** interact with one another
+![Sequence Diagram](images-dg/Sequence%20Diagram.png)
```

Comment:
![image](https://user-images.githubusercontent.com/28587601/97528532-f0eb6d80-19e8-11eb-8d71-d412e6a2a14f.png)

I believe that you should add a cross at the end of the life for the NormalCommand object. Furthermore, I believe from your logic class diagram below, you state that the parser is the one that creates the Command object. I take this to mean you instantiate a new Command object. In this case, you should point the arrow of execute to the constructor of the NormalCommand object.

This is found in your later diagrams.
![image](https://user-images.githubusercontent.com/28587601/97528712-550e3180-19e9-11eb-859e-9c30ee32b83d.png)

Reply:
@kaijiel24